### ISSUE TITLE
Suggestion add ModSecurity security Support

### ISSUE THREAD

**Issue by PHPJosh** [AnoopAlias/nDeploy#168]
Hi just want to drop this here but would like to suggest to either include the ModSecurity nginx module https://github.com/SpiderLabs/ModSecurity-nginx or make an extra RPM we can install to take advantage of the built in support on cpanel. as is using ModSecurity via cpanels integration is pointless unless we can get the module added here..  would make ModSecurity useful again with this plugin

**Comment by AnoopAlias** [comment]
@PHPJosh - This is planned as soon as ModSecurity version 3 has a stable release. I think the mod sec v3 rules are not compatible with the v2 ones which are supplied with cpanel httpd anyways. 

Also if you are using the PROXY to httpd mode whatever WAF you have configured on cPanel httpd will work .

**Comment by PHPJosh** [comment]
@AnoopAlias hmm well I have my server set to use Native Nginx and not proxying  as imo I think its better but once i start hosting sites will prob have to put a doc in my Knowledge base letting people know for now if they want to use ModSecurity to proxy to Apache. thanks Good to know the support is coming though :)

**Comment by AnoopAlias** [comment]
@PHPJosh - Yes native Nginx is better when it comes to performance. But if you want mod_sec support for now you need to use the PROXY mode .
mod_sec v3 is planned
The next version(in dev now) will have easy-switching capability to use NAXSI WAF with Drupal and Wordpress CMS that have community contributed whitelists

NAXSI is a high speed WAF and much better than mod_sec . But its not easy to configure as it needs extensive whitelisting of false positives to allow the genuine requests in and block the bad ones.

My Plan is to offer both NAXSI and Mod_sec(v3)

**Comment by PHPJosh** [comment]
@AnoopAlias hmm looking at NAXSI it does sound like its good also. but yea gotch bout the proxy mode.  think i will make it the default mode again once i start hosting other sites and just let people know they want more performance they will have to give up the mod_sec support for now

**Comment by PHPJosh** [comment]
@AnoopAlias also do you plan on implementing a WHM interface plugin or is there already one that i missed?

**Comment by AnoopAlias** [comment]
@PHPJosh - Its a cPanel plugin and no WHM interface exist. 

The idea is - Admin user install nginx and cPanel user have a choice between PROXY(works as if Xtendweb is not installed at all ) and native nginx mode and user will be able to switch back and forth at will. 

The admin user can alter the default mode(which is PROXY) etc by modifying default template etc .But these are all command line stuff and no GUI exist for this. http://xtendweb.gnusys.net/XtendWeb_Administration.html

In short as an admin you can install the plugin and let the end user play with the stuff and settle on what suits best for him.

**Comment by PHPJosh** [comment]
ah was just wondering about a GUI for us admins as when  we have access to a computer command line usage  is not a problem at all. as for one as admins if we are afraid of using ssh we have no business running a Web Host or servers for that matter.. but mainly a GUI would be useful if we are away from a PC as can quickly figure out issues with a touch interface. SSH over a mobile device is a pain in the butt. Maybe  i can look into making a WHM interface but will take alot of reading up on cpanel API as i have never made a Cpanel/WHM plugin lol. but of course this all just in my opinion but i am sure I am not the only one with this opinion.
